src/Widgets/AppContextMenu.vala

```diff
@@ -50,7 +53,15 @@ public class Slingshot.AppContextMenu : Gtk.Menu {
 #endif
 
     construct {
+
+        try {
+            switcheroo_control = new Slingshot.Backend.SwitcherooControl ();
```

Comment:
Same comment as before, I don't think this constructor can throw exceptions, so this try/catch probably not needed.